principles.md

```diff
@@ -0,0 +1,65 @@
+# Fejlesztési alapelvek
+## Együttműködés
+Mindenki külön branchen dolgozik.
+
+Feature branch név formátum: feature/{feature neve}/{fejlesztő neve}, pl. feature/auth/den.
+
+ - Miért? Azért, mert ha egy másik fejlesztő is be akar csatlakozni a feature fejlesztésébe, akkor nem lesznek random branch nevek a repoban, és az egyes branchek lényege egyértelmű lesz.
+
+Nem commitolunk más branchére. Helyette leágazunk belőle (a fentebb leírt branch név formátumot használva) és egy pull requestet indítunk, amiben meg lehet beszélni az implementációbeli különbségeket.
+
+## YAGNI & DRY
+YAGNI = You Ain't Gonna Need It
+
+DRY = Don't Repeat Yourself
+
+Előre dolgozni tilos! A kódismétlést elkerülő praktikák csak és kizárólag utólag alkalmazhatók!
+
+1. Make it work
+2. Make it right
+3. Make it fast
+
+---
+
+1. Fejleszted a featuret
+2. Működik, de a kód hányás
+3. Írsz hozzá teszteket
+4. Refaktorálod, hogy a kód ne legyen hányás. A működés most már a teszteknek köszönhetően biztosított.
+
+
+## Backend

```

Comment:
Ha a backendről is beszélünk itt, akkor ennek nem inkább a fő repóban kéne lennie?

Reply:
Majd áthelyezem, itt legyen előbb megreviewzva és bemergelve, de jogos.